11-bug_tracking.Rmd

```diff
@@ -1,23 +1,125 @@
+
+### Bug in the $\textsf{R}$ Language {#RLangBug}
+
+1. If there is a bug in the $\textsf{R}$ language itself, then you need to report using the same procedure and precautions as explained [above](#RCorePkgBug).
+
+### Bug in the non $\textsf{R}$ Core-supported packages and/or their documentations {#nonRCorePkgBug}
+
+1. For packages that are not maintained by the $\textsf{R}$ Core, the bug reports are usually submitted to their maintainer(s). You can find who is the package maintainer by running either of the commands `maintainer("package_name")` or `packageDescription("package_name")`.
+
+2. In $\textsf{R}$ the running function `bug.report(package = "package_name")` shall direct you to either the GitHub issue tracker of the package, or to the bug tracking web page, or towards composing an e-mail to the package maintainer. This function `bug.report` is disabled in $\textbf{RStudio}$.
+
+3. Please ensure that your bug is not already reported or fixed before reporting it in any of the ways suggested above.
+
+## Good practices in reporting bugs / Expectations of a (good) bug report
+
+If you follow the practices given below, you will come up with a good bug report which might make it easier for the Maintainer(s) to fix the bug.
+
+1. Include a minimal reproducible example of the bug in your report. The maintainer should be able to quickly reproduce the bug on using the minimal example that you provide.
+
+2. Mention the software architecture on which the bug occurred. 
+
+In addition to the above, here are the [bug writing guidelines](https://bugs.r-project.org/bugzilla/page.cgi?id=bug-writing.html) on Bugzilla. The [bug reporting](https://www.r-project.org/bugs.html#writing-a-good-bug-report) documentation in R also the discusses practices to write a good bug report.
+
+Once you have successfully reported a bug, you will likely receive an update each time an action is taken on the bug. On Bugzilla, the report may be given one of the following status: New, Assigned, Confirmed, Reopened, Unconfirmed.
+
+## Disagreement with a resolution on the bug tracker
+
+As humans, there might be differences of opinions from time to time. What needs to be considered here is that, we are respectful of the fact that care, thought, and volunteer time has gone into the resolution of the issue or bug. 
+
+If you take some time, then on reflection, the resolution steps may seem more reasonable than you initially thought. If you still feel that the resolution is incorrect, then raise a thoughtful question to the person who resolved it. If the issue was carefully thought about in the first place then is less likely to win any conversion of thought.
+
+As a reminder, issues closed by a core developer on the GitHub Mirror have already been carefully considered. Please do not reopen a closed issue.

```

Comment:
Perhaps mention that even if an issue or bug report is closed one can still comment on it.

Reply:
This should be Bugzilla vs GitHub mirror here.

Reply:
I have not read the whole document yet so it may be mentioned elsewhere, but every comment generates an email to every R-core member (or at least ones that don't have notifications disabled), and so care should be taken when commenting on issues in a couple of different cases, which include a) issues that have already been considered and closed, and b) commenting in pure agreement without adding anything beyond that to a discussion (+1 type posts which are perfectly acceptable in other contexts).